evm/src/cpu/kernel/asm/core/call.asm

```diff
@@ -4,6 +4,7 @@
 global sys_call:
     // stack: kexit_info, gas, address, value, args_offset, args_size, ret_offset, ret_size
     // TODO: Charge gas.
+    DUP3 %u256_to_addr %insert_accessed_addresses POP // TODO: Use return value in gas calculation.
```

Comment:
It looks like we were failing to truncate address arguments for a bunch of syscalls, and you've fixed most of it here, but here the original (untruncated) address argument would still be used by some call code further down. Would you mind fixing that also? (Like maybe `SWAP2 %u256_to_addr SWAP2` at the top.)